src/python/pants/backend/codegen/protobuf/python/python_protobuf_subsystem.py

```diff
@@ -118,52 +149,78 @@ async def infer_dependencies(
         )
         locality = source_root.path
 
-    addresses_for_protobuf = await Get(
-        PythonModuleOwners,
-        PythonModuleOwnersRequest(
-            "google.protobuf",
-            resolve=resolve,
-            locality=locality,
-        ),
-    )
-
-    result = [
-        find_python_runtime_library_or_raise_error(
-            addresses_for_protobuf,
-            request.field_set.address,
-            "google.protobuf",
-            resolve=resolve,
-            resolves_enabled=python_setup.enable_resolves,
-            recommended_requirement_name="protobuf",
-            recommended_requirement_url="https://pypi.org/project/protobuf/",
-            disable_inference_option=f"[{python_protobuf.options_scope}].infer_runtime_dependency",
-        )
-    ]
-
-    if request.field_set.grpc_toggle.value:
-        addresses_for_grpc = await Get(
+    result = []
+    if not request.field_set.grpc_toggle.value or python_protobuf.grpcio_plugin:
```

Comment:
Why do we not want the google.protobuf lib present if using grpclib? Does it supersede that as well as grpcio?

Reply:
Seems like you can use `grpclib` without protobuf, it's an optional dependency. Although now that I think about it again, it's probably because `grpclib` itself doesn't directly depends on protobuf, but if you are enabling this grpclib plugin with protobuf files, it's likely that you will need protobuf as well. Will remove this.